.gitlab-ci.yml

```diff
@@ -198,6 +243,12 @@ API:
         INTERNAL_NETWORK: ["true"]
         DISTRO_CODE: ["rhel_90"]
 
+API-nightly:
+  stage: test
+  extends: API
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "schedule" && $RUNNER =~ /[\S]+\/rhel-8\..-x86_64/'
+
```

Comment:
Grrr, found out this is missing the definitions for OStree jobs. 

Isn't there any better way to selectively execute a subset of the test matrix ? 

What would happen if you remove the section overrides and instead add the following rule to all sections:
```
- if: '$CI_PIPELINE_SOURCE != "schedule" || ($CI_PIPELINE_SOURCE == "schedule" && $RUNNER =~ /[\S]+\/rhel-8\..-x86_64/)'
```

**Note:** syntax may not be correct. What I want to say is "execute if not started by schedule or (started by schedule and the runner is RHEL)".

Reply:
That's a pretty good idea, I'll try it. It should be possible to define several rules per and execute the jobs under different conditions. This means that we also want to run OStree tests for nightly right?

Reply:
Yup, works as expected.
Regular pipeline: https://gitlab.com/osbuild/ci/osbuild-composer/-/pipelines/326103978
Scheduled pipeline: https://gitlab.com/osbuild/ci/osbuild-composer/-/pipelines/326104181

Unless you can spot some other potential problem I'll squash the latest commit. What I can see is that it won't have 'nightly' in the name but we can deduce that from the fact that it's going to be a scheduled pipeline and also from the stages as it won't have rpbuild. And then I'm not sure if it's immediately obvious from the code what it's supposed to do but I guess that's just gitlab-ci.yml syntax... I can add a comment though